models/certs.go

```diff
@@ -32,3 +34,51 @@ type CertInfo struct {
 	ValidTo   string    `json:"ValidTo"`
```

Comment:
The provided pull request adds a new set of types and functions for converting CSRConfig and CertInfo structs to their corresponding model types for use with the HashiCorp Terraform provider. Here are some suggestions for improvements:

1. Consistent naming conventions:
In the `NewCSRConfigModel` and `NewCertInfoModel` functions, the naming conventions are inconsistent. The function names are capitalized (CamelCase), while the variable names inside the functions are camelCase. It would be better to stick to a consistent naming convention throughout the code.

For example, in the `NewCSRConfigModel` function, the variable names `input.DistinguishedName`, `input.DepartmentName`, etc., should be changed to `distinguishedName`, `departmentName`, etc.

Here's an example of how the corrected functions could look:

```go
// NewCSRConfigModel - Converts CSRConfig to CSRConfigModel
func newCSRConfigModel(input CSRConfig) CSRConfigModel {
    return CSRConfigModel{
        distinguishedName: types.StringValue(input.distinguishedName),
        departmentName:    types.StringValue(input.departmentName),
        businessName:      types.StringValue(input.businessName),
        locality:          types.StringValue(input.locality),
        state:             types.StringValue(input.state),
        country:           types.StringValue(input.country),
        email:             types.StringValue(input.email),
        sans:              types.StringValue(input.sans),
    }
}
```

2. Type conversions:
In the `NewCSRConfigModel` and `NewCertInfoModel` functions, the code uses StringValue to convert the input fields to strings. This can be done more concisely using the String() method provided by the types package.

For example, instead of:

```go
returns CSRConfigModel{
    distinguishedName: types.StringValue(input.DistinguishedName),
    // ...
}
```

You can use:

```go
return CSRConfigModel{
    distinguishedName: types.String(input.DistinguishedName),
    // ...
}
```

This will avoid the need to manually convert the input fields to strings.

3. Error handling:
The provided code does not handle errors or edge cases. For example, what if the input structs are nil or have invalid values? You should consider adding error handling to the functions to ensure they can correctly handle unexpected input.

4. Code readability:
The code is generally easy to read, but some of the variable names could be more descriptive. For example, `Sans` could be renamed to `SubjectAlternateNames` to better convey its meaning.

Overall, the code is well-structured and follows the HashiCorp Terraform provider's naming conventions. With the suggested improvements, the code will be more readable, maintainable, and robust.